fix: GitHub API 403 errors in dependency managers due to unauthenticated rate limiting

## Describe the bug

Dependency managers that fetch release info from `api.github.com` fail intermittently with HTTP 403 when running on GitHub Actions. The affected managers are:

- `CraneDependencyManager` (`google/go-containerregistry`)
- `PodmanDependencyManager` (`containers/podman`)
- `GvproxyDependencyManager` (`containers/gvisor-tap-vsock`)
- `VfkitDependencyManager` (`crc-org/vfkit`)

Example failure from https://github.com/hiero-ledger/solo/actions/runs/27550183372/job/81436997322?pr=4650:

\`\`\`
DependencyInstallFailedSoloError: Failed to install dependencies: GitHub API request to https://api.github.com/repos/google/go-containerregistry/releases returned HTTP 403
  at Subcommand.installDependencies (src/core/command-path-builders/command-builder.ts:57:15)
Caused by: GitHubApiHttpResponseError: GitHub API request to https://api.github.com/repos/google/go-containerregistry/releases returned HTTP 403
  at CraneDependencyManager.fetchReleaseInfo (src/core/dependency-managers/crane-dependency-manager.ts:128:15)
\`\`\`

## Describe the expected behavior

Dependency installation should succeed without transient GitHub API rate limit errors on GitHub Actions runners.

## To Reproduce

Run multiple concurrent GitHub Actions workflows that trigger dependency installation (e.g., the one-shot podman test). Under load, the shared runner IP hits the 60 req/hour unauthenticated rate limit and GitHub returns 403.

## Additional Context

The root cause is that all four managers make **unauthenticated** GitHub API requests. GitHub's unauthenticated rate limit is 60 requests/hour per IP — shared across all concurrent jobs on the same runner pool. When multiple workflows run simultaneously, this quota is easily exhausted.

`network.ts` already applies the correct fix for CRD downloads (see lines 1248–1249): include `Authorization: Bearer $GITHUB_TOKEN` when the environment variable is set. GitHub Actions automatically provides `GITHUB_TOKEN` for every job, raising the effective limit to 1000 req/hour per installation token.

The fix is to apply the same pattern to the four affected dependency managers.

--- a/src/core/dependency-managers/crane-dependency-manager.ts
+++ b/src/core/dependency-managers/crane-dependency-manager.ts
@@ -116,12 +116,18 @@ export class CraneDependencyManager extends BaseDependencyManager {
 
   private async fetchReleaseInfo(tagName: string): Promise<ReleaseInfo> {
     try {
+      const headers: Record<string, string> = {
+        'User-Agent': constants.SOLO_USER_AGENT_HEADER,
+        Accept: 'application/vnd.github.v3+json',
+      };
+      // GITHUB_TOKEN is injected automatically into every GitHub Actions job,
+      // raising the rate limit from 60 req/hour (unauthenticated) to 1000 req/hour.
+      if (process.env.GITHUB_TOKEN) {
+        headers['Authorization'] = `Bearer ${process.env.GITHUB_TOKEN}`;
+      }
       const response: Response = await fetch(CRANE_RELEASES_LIST_URL, {
         method: 'GET',
-        headers: {
-          'User-Agent': constants.SOLO_USER_AGENT_HEADER,
-          Accept: 'application/vnd.github.v3+json',
-        },
+        headers,
       });
 
       if (!response.ok) {

--- a/src/core/dependency-managers/gvproxy-dependency-manager.ts
+++ b/src/core/dependency-managers/gvproxy-dependency-manager.ts
@@ -106,13 +106,16 @@ export class GvproxyDependencyManager extends BaseDependencyManager {
    */
   private async fetchReleaseInfo(tagName: string): Promise<ReleaseInfo> {
     try {
-      // Make a GET request to GitHub API using fetch
+      const headers: Record<string, string> = {
+        'User-Agent': constants.SOLO_USER_AGENT_HEADER,
+        Accept: 'application/vnd.github.v3+json',
+      };
+      if (process.env.GITHUB_TOKEN) {
+        headers['Authorization'] = `Bearer ${process.env.GITHUB_TOKEN}`;
+      }
       const response = await fetch(GVPROXY_RELEASES_LIST_URL, {
         method: 'GET',
-        headers: {
-          'User-Agent': constants.SOLO_USER_AGENT_HEADER,
-          Accept: 'application/vnd.github.v3+json', // Explicitly request GitHub API v3 format
-        },
+        headers,
       });
 
       if (!response.ok) {

--- a/src/core/dependency-managers/podman-dependency-manager.ts
+++ b/src/core/dependency-managers/podman-dependency-manager.ts
@@ -90,13 +90,16 @@ export class PodmanDependencyManager extends BaseDependencyManager {
    */
   private async fetchReleaseInfo(tagName: string): Promise<ReleaseInfo> {
     try {
-      // Make a GET request to GitHub API using fetch
+      const headers: Record<string, string> = {
+        'User-Agent': constants.SOLO_USER_AGENT_HEADER,
+        Accept: 'application/vnd.github.v3+json',
+      };
+      if (process.env.GITHUB_TOKEN) {
+        headers['Authorization'] = `Bearer ${process.env.GITHUB_TOKEN}`;
+      }
       const response = await fetch(PODMAN_RELEASES_LIST_URL, {
-        method: 'GET', // Changed from HEAD to GET to retrieve the body
-        headers: {
-          'User-Agent': constants.SOLO_USER_AGENT_HEADER,
-          Accept: 'application/vnd.github.v3+json', // Explicitly request GitHub API v3 format
-        },
+        method: 'GET',
+        headers,
       });
 
       if (!response.ok) {

--- a/src/core/dependency-managers/vfkit-dependency-manager.ts
+++ b/src/core/dependency-managers/vfkit-dependency-manager.ts
@@ -82,13 +82,16 @@ export class VfkitDependencyManager extends BaseDependencyManager {
    */
   private async fetchReleaseInfo(tagName: string): Promise<ReleaseInfo> {
     try {
-      // Make a GET request to GitHub API using fetch
+      const headers: Record<string, string> = {
+        'User-Agent': constants.SOLO_USER_AGENT_HEADER,
+        Accept: 'application/vnd.github.v3+json',
+      };
+      if (process.env.GITHUB_TOKEN) {
+        headers['Authorization'] = `Bearer ${process.env.GITHUB_TOKEN}`;
+      }
       const response = await fetch(VFKIT_RELEASES_LIST_URL, {
-        method: 'GET', // Changed from HEAD to GET to retrieve the body
-        headers: {
-          'User-Agent': constants.SOLO_USER_AGENT_HEADER,
-          Accept: 'application/vnd.github.v3+json', // Explicitly request GitHub API v3 format
-        },
+        method: 'GET',
+        headers,
       });
 
       if (!response.ok) {

